pkg/codegen/go/gen_program.go

```diff
@@ -598,19 +658,21 @@ func (g *generator) useLookupInvokeForm(token string) bool {
 	pkg, module, member, _ := hcl2.DecomposeToken(token, *new(hcl.Range))
 	modSplit := strings.Split(module, "/")
 	mod := modSplit[0]
-	if mod == "index" {
-		mod = ""
-	}
 	fn := Title(member)
-	if len(modSplit) >= 2 {
+	if mod == "index" && len(modSplit) >= 2 {
+		mod = ""
 		fn = Title(modSplit[1])
+	} else {
+		// e.g. for type ec2/getVpc:getVpcArgs
```

Comment:
This whole function is a little hard to reason about and could do with rewriting? The change here is necessary for invokes in the azure-nextgen provider which I tested manually as well.

Reply:
Yeah, I think this has accrued a lot of incremental complexity. For the time being, perhaps you could add a comment or two to document the case we're accommodating here. Every time I look back on this code it's like archeology to figure out what all of the special cases are for. Some context will be helpful if we do decide to do some refactoring and architectural improvements as a part of q4 debt collection.